In [1]:
import re, requests
from bs4 import BeautifulSoup
from collections import defaultdict
import json


In [2]:
reg = re.compile(r'center=(\d+\.\d+)%2C(\d+\.\d+)&')

In [3]:
place = ['Barwala School', 'Rohini Sec 18', 'Badli Crossing', 'Old Secretariat (Postal Account Office)', 'Tees Hazari Court', 'Sanjay Gandhi Transport Nagar', 'PNB STA Road', 'INS Hostel', 'Rohini Sec 24 Dear Satguru Sewaram Crossing', 'Rohini Sec-9 /7', 'Ashok Vihar Crossing', 'Bawana', 'Richi Rich Shalimar Bagh', 'Nirankari Colony', 'Jain Mandir', 'Prehladpur Village', 'Wazirabad Village (Burari)', 'Shahbad Daulatpur Gaon', 'GTK Depot', 'Delhi Engg College Shahbad', 'Bajitpur Mandir', 'Rohini West Metro Station', 'MCD Store (Rajpur Road)', 'Bawana School', 'Kohat Enclave Metro Station', 'ISBT Kashmere Gate Metro Station', 'DSIDC Bawana/ Dhakewala', 'Mall Road', 'Jawahar Navodya School', 'New Sabzi Mandi Azadpur', 'Bawana Soap Factory', 'Pitampura ND Block', 'Vishwa Vidyalaya Metro Station', 'Shahbad Daulatpur School', 'Rohini DTC Depot 4', 'Dhaka Village', 'Sec-24 Rohini Main Road', 'Jahangirpuri GT Road (Metro Station)', 'PS Bawana', 'GTB Nagar', 'Delhi Gate', 'Jain Colony', 'Nangli Poona (GT Road)', 'Nangal Thakran', 'Vaishali Enclave / Kapil Vihar', 'Maharishi Valmiki Hospital', 'Haiderpur Metro Station', 'Bajitpur Village', 'Lakhhi Ram Park', 'CV Raman ITI', 'Prehladpur Crossing', 'Exchange Store (Hill Road)', 'Barwala Village', 'Express Building / Shaheed Park (Bhagat Singh Terminal)', 'Madhuban Chowk', 'GPO', 'Police Station Begumpur', 'Azadpur Terminal', 'DSIDC Office Bawana', 'Libas Pur GT Road', 'Rohini Sec 35', 'Rithala Village', 'Katewara Village', 'Rohini Sec 23 Pocket 1 (Green Hill Aptt.)', 'Novelty Cinema', 'Poothkhurd Village', 'Rohini Sec 7 and 8 Crossing', 'Mandi House', 'TB Hospital', 'Rithala Metro Station', 'Badli Village', 'Alipur Village GT Road', 'Shivaji Stadium Terminal', 'Nirankari Sarover- Burari Xing', 'Shahbad Dairy A Block', 'Bawana Depot', 'Jind Pur Crossing', 'Mahindra Park', 'Katewara Crossing', 'Poothkalan More', 'Auchandi Border', 'Rohini Sec 27 and 30 Crossing', 'Shaheed Captain Vikram Batra Flyover', 'Poothkhurd Firni Road', 'Madhuban Chowk Pitampura Metro Stn', 'Rajpur Road Transport Deptt', 'Harewali Crossing', 'Aditi College Bawana', 'Vidhan Sabha Metro Station', 'Jharoda Metro Station', 'Pal Colony', 'Swaroop Nagar GT Road', 'Jama Masjid', 'Jail Road Rohini Sec 18', 'Balmiki Gandhi Ashram', 'Rohini Sec 20/21/22 Crossing', 'Model Town II', 'ISBT Nityanand Marg', 'Alpana Cinema (Model Town Mtr Stn)', 'Prem Bari Pull', 'Ludlow Castle', 'Red Fort', 'Guru Govind Singh University (Kashmere Gate)', 'Tilak Bridge', 'Shri Krishna Apartments', 'B.R. International School', 'Dariyapur Kalan School', 'Rohini Sec 22 Terminal', 'Rohini Sec-16 Xing', 'Wazirpur Depot / Subhash Place Depot', 'Scindia House', 'Rohini Sec 18 Pocket A', 'Rohini Sec-21', 'Exchange Store / Civil Lines Metro Station', 'Dariyapur Gaon', 'Gurudwara Siraspur GT Road', 'Rohini Sec 18 B Block', 'Shahbad Dairy', 'Mori Gate Terminal', 'Aruna Asaf Ali Hospital', 'Darya Ganj Golcha Cinema', 'Sarai Pipal Thala', 'Rohini Sec-9', 'Hanuman Mandir Rohini Sec 24', 'Auchandi Sarvodaya School', 'St. Xavier School Shahbad', 'Rohini Depot 1 Sec 6', 'Qutub Garh Village (T)', 'Mungeshpur Village', 'Bawana Dispensary', 'Prehladpur School', 'Bajit Pur Crossing', 'ITO', 'Sai Mandir', 'Model Town III', 'Kasturba Gandhi Road Crossing', 'Auchandi Village', 'Rohini Sec 23', 'Khyber Pass', 'Fire Station Sec-8', 'Sultan Pur Crossing / Poothkhurd', 'Adarsh Nagar Metro Station', 'Pitampura JD Block', 'Netaji Subhash Palace (Wazirpur Depot) Road No 41', 'Max Muller Bhawan', 'New Police Line', 'Dera Satguru Sewaram Sahai Sec 24/25 Crossing', 'Budhpur GT Road', 'Rohini Sec 18 B-7 Road', 'Mcd Store', 'Burari Metro Station', 'IP College', 'Old Delhi Railway Station', 'Radio Colony']

In [4]:
pos = {i+1:j for i,j in enumerate(place)}

pos

{1: 'Barwala School',
 2: 'Rohini Sec 18',
 3: 'Badli Crossing',
 4: 'Old Secretariat (Postal Account Office)',
 5: 'Tees Hazari Court',
 6: 'Sanjay Gandhi Transport Nagar',
 7: 'PNB STA Road',
 8: 'INS Hostel',
 9: 'Rohini Sec 24 Dear Satguru Sewaram Crossing',
 10: 'Rohini Sec-9 /7',
 11: 'Ashok Vihar Crossing',
 12: 'Bawana',
 13: 'Richi Rich Shalimar Bagh',
 14: 'Nirankari Colony',
 15: 'Jain Mandir',
 16: 'Prehladpur Village',
 17: 'Wazirabad Village (Burari)',
 18: 'Shahbad Daulatpur Gaon',
 19: 'GTK Depot',
 20: 'Delhi Engg College Shahbad',
 21: 'Bajitpur Mandir',
 22: 'Rohini West Metro Station',
 23: 'MCD Store (Rajpur Road)',
 24: 'Bawana School',
 25: 'Kohat Enclave Metro Station',
 26: 'ISBT Kashmere Gate Metro Station',
 27: 'DSIDC Bawana/ Dhakewala',
 28: 'Mall Road',
 29: 'Jawahar Navodya School',
 30: 'New Sabzi Mandi Azadpur',
 31: 'Bawana Soap Factory',
 32: 'Pitampura ND Block',
 33: 'Vishwa Vidyalaya Metro Station',
 34: 'Shahbad Daulatpur School',
 35: 'Rohini DTC

In [5]:
reg.findall("center=29.03880045%2C76.48792045&")

[('29.03880045', '76.48792045')]

In [6]:
coord = defaultdict(lambda :{"lat":None,"long":None})

In [7]:
special_symbols = {'/':"%2F",' ':"+"}

special_symbols

{'/': '%2F', ' ': '+'}

In [8]:
with open('route.json','r') as f:
    coord = json.load(f)

In [9]:
def clean_url(val):
    for i,j in special_symbols.items():
        val = val.replace(i,j)
    
    return val

In [10]:
def find_place(place):
    url = f'https://www.google.com/maps/search/{clean_url(place)}+bus+stop,+delhi,+india'
    res = requests.get(url=url)
    res.raise_for_status()

    soup = BeautifulSoup(res.text,"html.parser")
    link = str(soup.find('meta', {'itemprop': 'image'}))

    if link:
        return [j for i in map(lambda x: (float(x[0]),float(x[1])),reg.findall(link)) for j in i]
    else:
        return [None,None]

In [59]:
for i in range(133,len(place)+1):
    
    coord[pos[i]]["lat"], coord[pos[i]]["long"] = find_place(pos[i])
    
    print(f'{i} : {pos[i]} done => {coord[pos[i]]}, url => https://www.google.com/maps/search/{clean_url(pos[i])}+bus+stop,+delhi,+india')
        

133 : ITO done => {'lat': 28.628886, 'long': 77.2456055}, url => https://www.google.com/maps/search/ITO+bus+stop,+delhi,+india
134 : Sai Mandir done => {'lat': 28.6890107, 'long': 77.1141222}, url => https://www.google.com/maps/search/Sai+Mandir+bus+stop,+delhi,+india
135 : Model Town III done => {'lat': 28.70512085, 'long': 77.1879005}, url => https://www.google.com/maps/search/Model+Town+III+bus+stop,+delhi,+india
136 : Kasturba Gandhi Road Crossing done => {'lat': 28.6233675, 'long': 77.2266695}, url => https://www.google.com/maps/search/Kasturba+Gandhi+Road+Crossing+bus+stop,+delhi,+india
137 : Auchandi Village done => {'lat': 28.821976, 'long': 76.9939971}, url => https://www.google.com/maps/search/Auchandi+Village+bus+stop,+delhi,+india
138 : Rohini Sec 23 done => {'lat': 28.72089475, 'long': 77.0801056}, url => https://www.google.com/maps/search/Rohini+Sec+23+bus+stop,+delhi,+india
139 : Khyber Pass done => {'lat': 28.6904, 'long': 77.221285}, url => https://www.google.com/maps/

In [1]:
import datetime

datetime.datetime.now()

datetime.datetime(2024, 9, 9, 12, 13, 17, 871453)

In [11]:
print(coord)

{'Express Building / Shaheed Park (Bhagat Singh Terminal)': {'lat': 28.6400445, 'long': 77.13522}, 'Barwala School': {'lat': 28.7618325, 'long': 77.062134}, 'Rohini Sec 18': {'lat': 28.74033515, 'long': 77.1350395}, 'Badli Crossing': {'lat': 28.728556, 'long': 77.146155}, 'Old Secretariat (Postal Account Office)': {'lat': 28.6838315, 'long': 77.222193}, 'Tees Hazari Court': {'lat': 28.6668885, 'long': 77.2185865}, 'Sanjay Gandhi Transport Nagar': {'lat': 28.73766725, 'long': 77.1485955}, 'PNB STA Road': {'lat': 28.678422, 'long': 77.2207025}, 'INS Hostel': {'lat': 28.6961575, 'long': 77.21192}, 'Rohini Sec 24 Dear Satguru Sewaram Crossing': {'lat': 28.72827, 'long': 77.092445}, 'Rohini Sec-9 /7': {'lat': 28.708705, 'long': 77.1212473}, 'Ashok Vihar Crossing': {'lat': 28.701991, 'long': 77.164812}, 'Bawana': {'lat': 28.789049, 'long': 77.0454015}, 'Richi Rich Shalimar Bagh': {'lat': 28.7011835, 'long': 77.16282}, 'Nirankari Colony': {'lat': 28.7157968, 'long': 77.2058565}, 'Jain Mandir'

defaultdict(<function <lambda> at 0x0000022CC863CB80>, {'Express Building / Shaheed Park (Bhagat Singh Terminal)': {'lat': 28.6400445, 'long': 77.13522}, 'Barwala School': {'lat': 28.7618325, 'long': 77.062134}, 'Rohini Sec 18': {'lat': 28.74033515, 'long': 77.1350395}, 'Badli Crossing': {'lat': 28.728556, 'long': 77.146155}, 'Old Secretariat (Postal Account Office)': {'lat': 28.6838315, 'long': 77.222193}, 'Tees Hazari Court': {'lat': 28.6668885, 'long': 77.2185865}, 'Sanjay Gandhi Transport Nagar': {'lat': 28.73766725, 'long': 77.1485955}, 'PNB STA Road': {'lat': 28.678422, 'long': 77.2207025}, 'INS Hostel': {'lat': 28.6961575, 'long': 77.21192}, 'Rohini Sec 24 Dear Satguru Sewaram Crossing': {'lat': 28.72827, 'long': 77.092445}, 'Rohini Sec-9 /7': {'lat': 28.708705, 'long': 77.1212473}, 'Ashok Vihar Crossing': {'lat': 28.701991, 'long': 77.164812}, 'Bawana': {'lat': 28.789049, 'long': 77.0454015}, 'Richi Rich Shalimar Bagh': {'lat': 28.7011835, 'long': 77.16282}, 'Nirankari Colony': {'lat': 28.7157968, 'long': 77.2058565}, 'Jain Mandir': {'lat': 28.777437, 'long': 77.1423375}, 'Prehladpur Village': {'lat': 28.626357, 'long': 77.18512}, 'Wazirabad Village (Burari)': {'lat': 28.72373, 'long': 77.21661}, 'Shahbad Daulatpur Gaon': {'lat': 28.7452605, 'long': 77.1116925}, 'GTK Depot': {'lat': 28.731041, 'long': 77.158627}, 'Delhi Engg College Shahbad': {'lat': 28.74532, 'long': 77.11643}, 'Bajitpur Mandir': {'lat': 28.80338, 'long': 76.99795}, 'Rohini West Metro Station': {'lat': 28.7151345, 'long': 77.115092}, 'MCD Store (Rajpur Road)': {'lat': 28.6732645, 'long': 77.21685}, 'Bawana School': {'lat': 28.8030391, 'long': 77.0497101}, 'Kohat Enclave Metro Station': {'lat': 28.6973355, 'long': 77.1419926}, 'ISBT Kashmere Gate Metro Station': {'lat': 28.6678065, 'long': 77.229164}, 'DSIDC Bawana/ Dhakewala': {'lat': 28.7832415, 'long': 77.042485}, 'Mall Road': {'lat': 28.69978685, 'long': 77.20279155}, 'Jawahar Navodya School': {'lat': 28.70251055, 'long': 76.9379539}, 'New Sabzi Mandi Azadpur': {'lat': 28.7143, 'long': 77.1723975}, 'Bawana Soap Factory': {'lat': 28.80184, 'long': 77.03929}, 'Pitampura ND Block': {'lat': 28.6977075, 'long': 77.14556}, 'Vishwa Vidyalaya Metro Station': {'lat': 28.6951474, 'long': 77.21471015}, 'Shahbad Daulatpur School': {'lat': 28.745659, 'long': 77.1075555}, 'Rohini DTC Depot 4': {'lat': 28.744925, 'long': 77.130215}, 'Dhaka Village': {'lat': 28.67249, 'long': 77.1891505}, 'Sec-24 Rohini Main Road': {'lat': 28.728848, 'long': 77.0955205}, 'Jahangirpuri GT Road (Metro Station)': {'lat': 28.72593, 'long': 77.16292}, 'PS Bawana': {'lat': 28.80653, 'long': 77.023035}, 'GTB Nagar': {'lat': 28.6987871, 'long': 77.20402655}, 'Delhi Gate': {'lat': 28.6061465, 'long': 77.22228545}, 'Jain Colony': {'lat': 28.7265115, 'long': 77.0502455}, 'Nangli Poona (GT Road)': {'lat': 28.7731585, 'long': 77.1414578}, 'Nangal Thakran': {'lat': 28.8089135, 'long': 77.007915}, 'Vaishali Enclave / Kapil Vihar': {'lat': 28.699538, 'long': 77.137516}, 'Maharishi Valmiki Hospital': {'lat': 28.775105, 'long': 77.047525}, 'Haiderpur Metro Station': {'lat': 28.73096, 'long': 77.149795}, 'Bajitpur Village': {'lat': 28.807505, 'long': 77.004395}, 'Lakhhi Ram Park': {'lat': 28.7133205, 'long': 77.06059}, 'CV Raman ITI': {'lat': 28.7238585, 'long': 77.2022595}, 'Prehladpur Crossing': {'lat': 28.6325115, 'long': 77.1757905}, 'Exchange Store (Hill Road)': {'lat': 28.677785, 'long': 77.22304}, 'Barwala Village': {'lat': 28.760327, 'long': 77.0606425}, 'Madhuban Chowk': {'lat': 28.70366, 'long': 77.13055}, 'GPO': {'lat': 28.6619014, 'long': 77.2348118}, 'Police Station Begumpur': {'lat': 28.72274, 'long': 77.07137}, 'Azadpur Terminal': {'lat': 28.707595, 'long': 77.179035}, 'DSIDC Office Bawana': {'lat': 28.78715425, 'long': 77.04072465}, 'Libas Pur GT Road': {'lat': 28.75137, 'long': 77.150065}, 'Rohini Sec 35': {'lat': 28.7574065, 'long': 77.0668175}, 'Rithala Village': {'lat': 28.720285, 'long': 77.1022125}, 'Katewara Village': {'lat': 28.80513175, 'long': 76.97683325}, 'Rohini Sec 23 Pocket 1 (Green Hill Aptt.)': {'lat': 28.71925, 'long': 77.084485}, 'Novelty Cinema': {'lat': 28.661455, 'long': 77.221072}, 'Poothkhurd Village': {'lat': 28.7705925, 'long': 77.05186995}, 'Rohini Sec 7 and 8 Crossing': {'lat': 28.7083005, 'long': 77.1253316}, 'Mandi House': {'lat': 28.6253402, 'long': 77.23217985}, 'TB Hospital': {'lat': 28.61450675, 'long': 77.19747205}, 'Rithala Metro Station': {'lat': 28.721601, 'long': 77.1061947}, 'Badli Village': {'lat': 28.73514855, 'long': 77.1415215}, 'Alipur Village GT Road': {'lat': 28.7973155, 'long': 77.133467}, 'Shivaji Stadium Terminal': {'lat': 28.632669, 'long': 77.2157855}, 'Nirankari Sarover- Burari Xing': {'lat': 28.7274581, 'long': 77.1979306}, 'Shahbad Dairy A Block': {'lat': 28.745854, 'long': 77.0985165}, 'Bawana Depot': {'lat': 28.802645, 'long': 77.05219}, 'Jind Pur Crossing': {'lat': 28.7933605, 'long': 77.1442015}, 'Mahindra Park': {'lat': 28.702446, 'long': 77.1510865}, 'Katewara Crossing': {'lat': 28.8061, 'long': 76.96585}, 'Poothkalan More': {'lat': 28.7177055, 'long': 77.0775405}, 'Auchandi Border': {'lat': 28.8288445, 'long': 76.9886015}, 'Rohini Sec 27 and 30 Crossing': {'lat': 28.748642, 'long': 77.089265}, 'Shaheed Captain Vikram Batra Flyover': {'lat': 28.73492, 'long': 77.156308}, 'Poothkhurd Firni Road': {'lat': 28.768395, 'long': 77.0519015}, 'Madhuban Chowk Pitampura Metro Stn': {'lat': 28.70231, 'long': 77.13337}, 'Rajpur Road Transport Deptt': {'lat': 28.6791917, 'long': 77.218635}, 'Harewali Crossing': {'lat': 28.824916, 'long': 77.005239}, 'Aditi College Bawana': {'lat': 28.794485, 'long': 77.036512}, 'Vidhan Sabha Metro Station': {'lat': 28.68772865, 'long': 77.2216628}, 'Jharoda Metro Station': {'lat': 28.6731743, 'long': 77.0692407}, 'Pal Colony': {'lat': 28.723579, 'long': 77.103647}, 'Swaroop Nagar GT Road': {'lat': 28.75807375, 'long': 77.1478755}, 'Jama Masjid': {'lat': 28.651042, 'long': 77.2381395}, 'Jail Road Rohini Sec 18': {'lat': 28.7274, 'long': 77.1433055}, 'Balmiki Gandhi Ashram': {'lat': 28.704864, 'long': 77.205101}, 'Rohini Sec 20/21/22 Crossing': {'lat': 28.7202505, 'long': 77.0692375}, 'Model Town II': {'lat': 28.70414, 'long': 77.190575}, 'ISBT Nityanand Marg': {'lat': 28.668838, 'long': 77.226404}, 'Alpana Cinema (Model Town Mtr Stn)': {'lat': 28.702865, 'long': 77.193775}, 'Prem Bari Pull': {'lat': 28.6988615, 'long': 77.157863}, 'Ludlow Castle': {'lat': 28.6723515, 'long': 77.2259155}, 'Red Fort': {'lat': 28.6229811, 'long': 77.22888875}, 'Guru Govind Singh University (Kashmere Gate)': {'lat': 28.66542, 'long': 77.23019}, 'Tilak Bridge': {'lat': 28.6248195, 'long': 77.239452}, 'Shri Krishna Apartments': {'lat': 28.743881, 'long': 77.126152}, 'B.R. International School': {'lat': 28.8215505, 'long': 76.978604}, 'Dariyapur Kalan School': {'lat': 28.8089475, 'long': 77.0149275}, 'Rohini Sec 22 Terminal': {'lat': 28.71803415, 'long': 77.063905}, 'Rohini Sec-16 Xing': {'lat': 28.7412405, 'long': 77.1194835}, 'Wazirpur Depot / Subhash Place Depot': {'lat': 28.6933815, 'long': 77.1528615}, 'Scindia House': {'lat': 28.628698, 'long': 77.2219261}, 'Rohini Sec 18 Pocket A': {'lat': 28.741861, 'long': 77.1330545}, 'Rohini Sec-21': {'lat': 28.715124, 'long': 77.0664795}, 'Exchange Store / Civil Lines Metro Station': {'lat': 28.6766995, 'long': 77.22494}, 'Dariyapur Gaon': {'lat': 28.82061365040768, 'long': 77.01128065356461}, 'Gurudwara Siraspur GT Road': {'lat': 28.76491, 'long': 77.145725}, 'Rohini Sec 18 B Block': {'lat': 28.7357925, 'long': 77.137574}, 'Shahbad Dairy': {'lat': 28.74588285, 'long': 77.104108}, 'Mori Gate Terminal': {'lat': 28.667291, 'long': 77.226239}, 'Aruna Asaf Ali Hospital': {'lat': 28.66961, 'long': 77.21753}, 'Darya Ganj Golcha Cinema': {'lat': 28.645268, 'long': 77.240405}, 'Sarai Pipal Thala': {'lat': 28.720006, 'long': 77.167534}, 'Rohini Sec-9': {'lat': 28.7122885, 'long': 77.119985}, 'Hanuman Mandir Rohini Sec 24': {'lat': 28.7229335, 'long': 77.0876525}, 'Auchandi Sarvodaya School': {'lat': 28.8203015, 'long': 76.9976395}, 'St. Xavier School Shahbad': {'lat': 28.745643, 'long': 77.103037}, 'Rohini Depot 1 Sec 6': {'lat': 28.716215, 'long': 77.11295215}, 'Qutub Garh Village (T)': {'lat': 28.8010605, 'long': 76.961974}, 'Mungeshpur Village': {'lat': 28.8103686, 'long': 77.06059775}, 'Bawana Dispensary': {'lat': 28.80094, 'long': 77.037245}, 'Prehladpur School': {'lat': 28.752642, 'long': 77.0758505}, 'Bajit Pur Crossing': {'lat': 28.798305, 'long': 77.017495}, 'ITO': {'lat': 28.628886, 'long': 77.2456055}, 'Sai Mandir': {'lat': 28.6890107, 'long': 77.1141222}, 'Model Town III': {'lat': 28.70512085, 'long': 77.1879005}, 'Kasturba Gandhi Road Crossing': {'lat': 28.6233675, 'long': 77.2266695}, 'Auchandi Village': {'lat': 28.821976, 'long': 76.9939971}, 'Rohini Sec 23': {'lat': 28.72089475, 'long': 77.0801056}, 'Khyber Pass': {'lat': 28.6904, 'long': 77.221285}, 'Fire Station Sec-8': {'lat': 28.706946, 'long': 77.1268413}, 'Sultan Pur Crossing / Poothkhurd': {'lat': 28.767895, 'long': 77.049345}, 'Adarsh Nagar Metro Station': {'lat': 28.717531, 'long': 77.170037}, 'Pitampura JD Block': {'lat': 28.7016815, 'long': 77.134483}, 'Netaji Subhash Palace (Wazirpur Depot) Road No 41': {'lat': 28.69415, 'long': 77.15103}, 'Max Muller Bhawan': {'lat': 28.6244435, 'long': 77.2245335}, 'New Police Line': {'lat': 28.69701435, 'long': 77.15506015}, 'Dera Satguru Sewaram Sahai Sec 24/25 Crossing': {'lat': 28.72976, 'long': 77.094635}, 'Budhpur GT Road': {'lat': 28.781983, 'long': 77.14136}, 'Rohini Sec 18 B-7 Road': {'lat': 28.742964, 'long': 77.131826}, 'Mcd Store': {'lat': 28.6440836, 'long': 77.09323125}, 'Burari Metro Station': {'lat': 28.6440836, 'long': 77.09323125}, 'IP College': {'lat': 28.68088, 'long': 77.223917}, 'Old Delhi Railway Station': {'lat': 28.6599985, 'long': 77.228309}, 'Radio Colony': {'lat': 28.710959, 'long': 77.2043855}})

{'Express Building / Shaheed Park (Bhagat Singh Terminal)': {'lat': None, 'long': None}, 'Barwala School': {'lat': 28.7618325, 'long': 77.062134}, 'Rohini Sec 18': {'lat': 28.74033515, 'long': 77.1339595}, 'Badli Crossing': {'lat': 28.728556, 'long': 77.146155}, 'Old Secretariat (Postal Account Office)': {'lat': 28.6838315, 'long': 77.222193}, 'Tees Hazari Court': {'lat': 28.6668885, 'long': 77.2185865}, 'Sanjay Gandhi Transport Nagar': {'lat': 28.71784325, 'long': 77.11863785}, 'PNB STA Road': {'lat': 28.678422, 'long': 77.2207025}, 'INS Hostel': {'lat': 28.6961575, 'long': 77.21192}, 'Rohini Sec 24 Dear Satguru Sewaram Crossing': {'lat': 28.72827, 'long': 77.092445}, 'Rohini Sec-9 /7': {'lat': 28.708705, 'long': 77.1196665}, 'Ashok Vihar Crossing': {'lat': 28.701915, 'long': 77.16475225}, 'Bawana': {'lat': 28.789049, 'long': 77.0454015}, 'Richi Rich Shalimar Bagh': {'lat': 28.7011835, 'long': 77.16282}, 'Nirankari Colony': {'lat': 28.714846, 'long': 77.204121}, 'Jain Mandir': {'lat': 28.777437, 'long': 77.1423375}, 'Prehladpur Village': {'lat': 28.626357, 'long': 77.18512}, 'Wazirabad Village (Burari)': {'lat': 28.72373, 'long': 77.21661}, 'Shahbad Daulatpur Gaon': {'lat': 28.7452605, 'long': 77.1116925}, 'GTK Depot': {'lat': 28.731041, 'long': 77.158627}, 'Delhi Engg College Shahbad': {'lat': 28.74532, 'long': 77.11643}}

In [12]:
coord[min(coord,key=lambda x:coord[x]["long"])],sorted(coord,key=lambda x:coord[x]["lat"])

({'lat': 28.70251055, 'long': 76.9379539},
 ['Delhi Gate',
  'TB Hospital',
  'Red Fort',
  'Kasturba Gandhi Road Crossing',
  'Max Muller Bhawan',
  'Tilak Bridge',
  'Mandi House',
  'Prehladpur Village',
  'Scindia House',
  'ITO',
  'Prehladpur Crossing',
  'Shivaji Stadium Terminal',
  'Express Building / Shaheed Park (Bhagat Singh Terminal)',
  'Mcd Store',
  'Burari Metro Station',
  'Darya Ganj Golcha Cinema',
  'Jama Masjid',
  'Old Delhi Railway Station',
  'Novelty Cinema',
  'GPO',
  'Guru Govind Singh University (Kashmere Gate)',
  'Tees Hazari Court',
  'Mori Gate Terminal',
  'ISBT Kashmere Gate Metro Station',
  'ISBT Nityanand Marg',
  'Aruna Asaf Ali Hospital',
  'Ludlow Castle',
  'Dhaka Village',
  'Jharoda Metro Station',
  'MCD Store (Rajpur Road)',
  'Exchange Store / Civil Lines Metro Station',
  'Exchange Store (Hill Road)',
  'PNB STA Road',
  'Rajpur Road Transport Deptt',
  'IP College',
  'Old Secretariat (Postal Account Office)',
  'Vidhan Sabha Metro Stat

In [69]:
with open('route.json','w') as f:
    json.dump(coord,f)

In [13]:
coord['Dariyapur Gaon'] = {'lat': 28.82061365040768,'long': 77.01128065356461}

In [14]:
f="Express Building / Shaheed Park (Bhagat Singh Terminal)"

find_place(f),print(f'{f} done => {coord[f]}, url => https://www.google.com/maps/search/{clean_url(f)}+bus+stop,+delhi,+india')

HTTPError: 429 Client Error: Too Many Requests for url: https://www.google.com/sorry/index?continue=https://www.google.com/maps/search/Express%2BBuilding%2B%252F%2BShaheed%2BPark%2B(Bhagat%2BSingh%2BTerminal)%2Bbus%2Bstop,%2Bdelhi,%2Bindia&q=EhAkBQIBAC5AeWGEPCPfW3CHGN3V9rYGIjDBRpgnu2msVbA6A9E50-HvZT5hfWOcQnJvALz2hsMbQkjiIgPhBEabB391MvrNC_wyAXJaAUM